core/process.py

```diff
@@ -1019,6 +1020,8 @@ def StartProcess(self, why):
         for st in self.state_changes:
             st.ApplyFromParent(self)
 
+        self.pgid = posix.getpgid(pid)
```

Comment:
potential race here: if the child wakes up before the parent and exits before this call, this might not work...

Reply:
probably very unlikely and not worth worrying about?

Reply:
decided relying on kernel state here is dangerous and added some of our own state to track whether procs are in the fg or bg